state/src/applicator/task_queue.rs

```diff
@@ -74,11 +74,8 @@ impl StateApplicator {
             self.maybe_start_task(task, &tx)?;
         }
 
-        // If the queue is empty and the task was a wallet task, run the matching engine
-        // on all orders that are ready
-        // TODO: We should only have one node execute the internal matching engine,
-        // though this is okay for the moment
-        if tasks.is_empty() && task.descriptor.is_wallet_task() {
+        if Self::should_run_matching_engine(&tasks, &task, &tx)? {
```

Comment:
nit: can we rename `tasks` to `remaining_tasks` or something? `tasks` next to `task` reads a little odd